src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp

```diff
@@ -1782,4 +1782,78 @@ void C2_MacroAssembler::compare_floating_point_v(VectorRegister vd, BasicType bt
       assert(false, "unsupported compare condition");
       ShouldNotReachHere();
   }
+}
+
+void C2_MacroAssembler::vector_integer_extend(VectorRegister dst, BasicType dst_bt, int length_in_bytes,
+                                              VectorRegister src, BasicType src_bt) {
+  assert(type2aelembytes(dst_bt) > type2aelembytes(src_bt) && type2aelembytes(dst_bt) <= 8 && type2aelembytes(src_bt) <= 4, "invalid element size");
+  assert(dst_bt != T_FLOAT && dst_bt != T_DOUBLE && src_bt != T_FLOAT && src_bt != T_DOUBLE, "should be integer element");
+  // https://github.com/riscv/riscv-v-spec/blob/master/v-spec.adoc#52-vector-operands
+  // The destination EEW is greater than the source EEW, the source EMUL is at least 1,
+  // and the overlap is in the highest-numbered part of the destination register group.
+  // Since LMUL=1, vd and vs cannot be the same.
+  assert_different_registers(dst, src);
+
+  rvv_vsetvli(dst_bt, length_in_bytes);
```

Comment:
I think we should use the actual AVL instread of 'length_in_bytes' for rvv_vsetvli ?

Reply:
No problem here, https://github.com/openjdk/jdk/blob/3554e7a3ffb879c7e5ef7547eb053e484d09d12b/src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp#L1830 There was a problem here and it has been fixed.